engine/apps/metrics_exporter/tasks.py

```diff
@@ -111,38 +114,72 @@ def calculate_and_cache_metrics(organization_id, force=False):
     }
 
     for integration in integrations:
+        metric_alert_group_total_data = {
+            "integration_name": integration.emojized_verbal_name,
+            "team_name": integration.team_name,
+            "team_id": integration.team_id_or_no_team,
+            "org_id": instance_org_id,
+            "slug": instance_slug,
+            "id": instance_id,
+            "services": {
+                NO_SERVICE_VALUE: get_default_states_dict(),
+            },
+        }
         # calculate states
         for state, alert_group_filter in states.items():
-            metric_alert_group_total.setdefault(
-                integration.id,
-                {
-                    "integration_name": integration.emojized_verbal_name,
-                    "team_name": integration.team_name,
-                    "team_id": integration.team_id_or_no_team,
-                    "org_id": instance_org_id,
-                    "slug": instance_slug,
-                    "id": instance_id,
-                },
-            )[state] = integration.alert_groups.filter(alert_group_filter).count()
-
-        # get response time
-        all_response_time = integration.alert_groups.filter(
-            started_at__gte=response_time_period,
-            response_time__isnull=False,
-        ).values_list("response_time", flat=True)
-
-        all_response_time_seconds = [int(response_time.total_seconds()) for response_time in all_response_time]
-
-        metric_alert_group_response_time[integration.id] = {
+            # count alert groups with `service_name` label group by label value
```

Comment:
Have we tried these queries on production db? Do we need to apply any optimizations?

Reply:
> +1 to optimize as possible (using the index improved things?)

the slowest call from my tests is ~10s now. It uses readonly db, so I don't think this would be a problem 🤔 

> n any case, this shouldn't affect the response time in the scrape endpoint right? (there we are still getting the data from cache and returning it, correct?)

Correct, we get all data from the cache and don't do any calculations there